setup.py

```diff
@@ -267,12 +258,12 @@ def finalize_options(self):
     def run(self):
         super().run()
 
-        if not self.distribution.no_update_icon_cache:
+        if not self.distribution.no_update_icon_cache and shutil.which("gtk-update-icon-cache"):
```

Comment:
Auto-magic is bad for distributions in that it needs to be found to not produce bad results without knowing and still end up as a hard requirement; better letter this crash or fail with an error please rather than not doing anything, silently. Same for the fix below. Thank you!

Reply:
yea that's perfectly reasonable, I didn't know if this was a hard requirement or not